src/TritonCTS/src/DbWrapper.cpp

```diff
@@ -74,25 +77,60 @@ void DbWrapper::initDB() {
 
 void DbWrapper::initAllClocks() {
         std::cout << " Initializing clock nets\n";
-        
+
+        clearNumClocks();
+
+        std::cout << " Looking for clock nets in the design\n";
+
+        //Uses dbSta to find all clock nets in the design.
+
+        std::set<odb::dbNet*> clockNets;
+
+        _openSta->findClkNets(clockNets);
+
+        //Checks the user input in case there are other nets that need to be added to the set.
         std::vector<std::string> clockNetNames;
```

Comment:
why are there clock net names? where are they coming from?

Reply:
These are only utilized when the user uses the "-clk_nets" parameter for TritonCTS. That parameter requires a string for the clock nets (the divisor being a space). This is transformed into a vector of strings by parseClockNames and then into each respective net in "odb::dbNet* net = _block->findNet(name.c_str());".

I'm not sure if this parameter is used on jenkins or some benchmark. It is used by some unit tests.

Reply:
I think the point is that it would be better to fix code to eliminate the need to parse the clock names. In net there is a net->get_name() being passed into a block->find_Net(). This is slow but also may not always work if the naming conventions do not align. If you can modify the code to accept a vector of dbNet* instead of strings it would simplify the code and make it more reliable.